jwst/assign_wcs/miri.py

```diff
@@ -74,53 +74,128 @@ def imaging_distortion(input_model, reference_files):
     ref_file: filter_offset.asdf - (1)
     ref_file: distortion.asdf -(2,3,4)
     """
+
+    # Load in the distortion file.
     distortion = AsdfFile.open(reference_files['distortion']).tree['model']
-    filter_offset = AsdfFile.open(reference_files['filteroffset']).tree[input_model.meta.instrument.filter]
-    full_distortion = models.Shift(filter_offset['row_offset']) & models.Shift(
-        filter_offset['column_offset']) | distortion
+
+    # ToDo: This will likely have to change in the future, but the "filteroffset" file we have
+    # ToDo: currently does not contain that key.
+    filter_offset = None
+    if input_model.meta.instrument.filter in  AsdfFile.open(reference_files['filteroffset']).tree:
+        filter_offset = AsdfFile.open(reference_files['filteroffset']).tree[input_model.meta.instrument.filter]
+        full_distortion = models.Shift(filter_offset['row_offset']) & models.Shift(
+            filter_offset['column_offset']) | distortion
+    else:
+        full_distortion = distortion
+
     full_distortion = full_distortion.rename('distortion')
+
     return full_distortion
 
 
+def is_number(s):
+
+    # Check, first, to make sure it is not None
+    if not s:
+        return False
+
+    # Now, check to see if it is a number
+    try:
+        float(s)
+        return True
+    except ValueError:
+        return False
+
 def lrs(input_model, reference_files):
     """
     Create the WCS pipeline for a MIRI fixed slit observation.
 
-    reference_files = {"specwcs": 'MIRI_FM_MIRIMAGE_P750L_DISTORTION_04.02.00.fits'}
+    reference_files = {
+        "specwcs": 'MIRI_FM_MIRIMAGE_P750L_DISTORTION_04.02.00.fits'
```

Comment:
You should be aware that there was an error in the header keyword values of the original version of this reference file as delivered by the MIRI IDT in CDP-4. In particular, IMXSLTL=60, which is the x-axis zeropoint of the slitless trace, is wrong (by a lot). The correct value is 34. The other values for IMX and IMY were also updated (by small amounts) in the more recent CDP-6 delivery of this ref file. The latest file can be found at /grp/jwst/wit/miri/CDP-6/CALSPEC2_LRS/distortion_wavelength_calibration/DISTORTION/referencefile/MIRI_FM_MIRIMAGE_P750L_DISTORTION_06.02.00.fits